src/Compiler/TaskModel.h

```diff
@@ -44,6 +44,10 @@ class TaskModel : public QAbstractTableModel {
   TaskManager *taskManager() const;
   void setTaskManager(TaskManager *newTaskManager);
 
+ signals:
+  void rowVisibilityChanged(const QModelIndex &topLeft,
```

Comment:
this is not how it should work. It should go over dataChange(). The question is why it doesn't work with dataChange?

Reply:
dataChanged() was emitted only for children of collapsed task, while the parent wasn't in that range and remained not repainted. 
including parent into dataChanged() didn't work because view() immediately hides it. so I added the signal specific to that action, while dataChanged() is now handled only by Qt.

Reply:
Include parent should work. Why it doesn't work? what do you mean view() immediately hides it. ?

Reply:
On the other hand, we can leave everything as it was and only emit additional layoutChanged() signal, which will make sure everything is repainted. it should solve both issues.